hack/lib/start.sh

```diff
@@ -0,0 +1,496 @@
+	echo "[INFO] Test artifacts will be in: ${ARTIFACT_DIR}"
+	echo "[INFO] Config dir is:		${SERVER_CONFIG_DIR}"
+	echo "[INFO] Using images:		${USE_IMAGES}"
+	echo "[INFO] MasterIP is:		${MASTER_ADDR}"
+
+	mkdir -p ${LOG_DIR}
+
+	echo "[INFO] Scan of OpenShift related processes already up via ps -ef	| grep openshift : "
+	ps -ef | grep openshift
+	echo "[INFO] Starting OpenShift server"
+	local openshift_env=( "OPENSHIFT_PROFILE=web" "OPENSHIFT_ON_PANIC=crash" )
+	$(os::start::internal::openshift_executable) start master \
+		--config=${MASTER_CONFIG_DIR}/master-config.yaml \
+		--loglevel=4 --logspec='*importer=5' \
+	&>"${LOG_DIR}/openshift.log" &
+	export OS_PID=$!
+
+	echo "[INFO] OpenShift server start at: "
+	date
+
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz" "apiserver: " 0.25 160
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz/ready" "apiserver(ready): " 0.25 160
+
+	echo "[INFO] OpenShift server health checks done at: "
+	date
+}
+readonly -f os::start::master
+
+# os::start::internal::start_all_in_one starts the OpenShift server all in one.
+# It is advised to use this function after a successful run of 'configure_os_server'
+#
+# Globals:
+#  - USE_SUDO
+#  - LOG_DIR
+#  - ARTIFACT_DIR
+#  - VOLUME_DIR
+#  - SERVER_CONFIG_DIR
+#  - USE_IMAGES
+#  - MASTER_ADDR
+#  - MASTER_CONFIG_DIR
+#  - NODE_CONFIG_DIR
+#  - API_SCHEME
+#  - API_HOST
+#  - API_PORT
+#  - KUBELET_SCHEME
+#  - KUBELET_HOST
+#  - KUBELET_PORT
+# Returns:
+#  - export OS_PID
+function os::start::internal::start_all_in_one () {
+	local use_latest_images
+	if [[ -n "${USE_LATEST_IMAGES:-}" ]]; then
+		use_latest_images="true"
+	else
+		use_latest_images="false"
+	fi
+
+	echo "[INFO] Starting OpenShift server"
+	local openshift_env=( "OPENSHIFT_PROFILE=web" "OPENSHIFT_ON_PANIC=crash" )
+	$(os::start::internal::openshift_executable) start \
+		--master-config=${MASTER_CONFIG_DIR}/master-config.yaml \
+		--node-config=${NODE_CONFIG_DIR}/node-config.yaml \
+		--loglevel=4 --logspec='*importer=5' \
+		--latest-images="${use_latest_images}" \
+	&>"${LOG_DIR}/openshift.log" &
+	export OS_PID=$!
+
+	echo "[INFO] OpenShift server start at: "
+	date
+
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz" "apiserver: " 0.25 80
+	wait_for_url "${KUBELET_SCHEME}://${KUBELET_HOST}:${KUBELET_PORT}/healthz" "[INFO] kubelet: " 0.5 120
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/healthz/ready" "apiserver(ready): " 0.25 80
+	wait_for_url "${API_SCHEME}://${API_HOST}:${API_PORT}/api/v1/nodes/${KUBELET_HOST}" "apiserver(nodes): " 0.25 80
+
+	echo "[INFO] OpenShift server health checks done at: "
+	date
+}
+readonly -f os::start::internal::start_all_in_one
+
+# os::start::internal::start_etcd starts etcd for OpenShift
+# Globals:
+#  - USE_SUDO
+#  - LOG_DIR
+#  - MASTER_CONFIG_DIR
+#  - API_SCHEME
+#  - API_HOST
+#  - ETCD_PORT
+# Arguments:
+#  None
+# Returns:
+#  - export ETCD_PID
+function os::start::internal::start_etcd() {
```

Comment:
I'm not sure anything but helper functions need to be internal. All of these could just be `os::start::etcd` *etc*